service-base/src/main/java/org/eclipse/hono/service/AbstractProtocolAdapterBase.java

```diff
@@ -719,8 +744,9 @@ protected void doStop(final Promise<Void> stopPromise) {
      *         connection cannot be established.
      * @throws NullPointerException if serviceName is {@code null}.
      * @throws IllegalArgumentException if factory is {@code null}.
+     * @param <T> The type of connection that the factory uses.
      */
-    protected final Future<HonoConnection> connectToService(final ConnectionLifecycle<HonoConnection> factory, final String serviceName) {
+    protected final <T> Future<T> connectToService(final ConnectionLifecycle<T> factory, final String serviceName) {
```

Comment:
I think the type parameter `T` should better be renamed so that it doesn't hide the `T` parameter of the enclosing class.